### PR TITLE
DDF-3860 PaosInInterceptor now forwards headers

### PR DESCRIPTION
**Pull request by rzwiefel**
#### What does this PR do?
Previously, the PaosInInterceptor would not forward the original headers
which would cause things to break. Now we add the original headers back in
as would be expected of a request that was not intercepted at all.

#### Who is reviewing it? 
anyone @clockard @stustison 

#### Select relevant component teams: 
@codice/security 

#### How should this be tested? (List steps with links to updated documentation)
Setup Two DDFs via CSW with IDP enabled as the authentication mechanism, ensure you can download a product from DDF B where the product was ingested into DDF A. Also it may not hurt to enable fanout proxy on  DDF B and ensure you can still download the product from there. 

#### Any background context you want to provide?
the `PaosInInterceptor` was not forwarding the correct headers causing things like ECP to not function correctly. 

#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-3860

#### Checklist:
- [na] Documentation Updated
- [x] Update / Add Unit Tests
- [na] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: platform/security/rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/paos/PaosInInterceptor.java [hunk 1]
```diff
@@ -243,6 +246,18 @@ public void handleMessage(Message message) throws Fault {
       if (httpResponse.statusCode < 400) {
         httpResponseContent = httpResponse.content;
         message.setContent(InputStream.class, httpResponseContent);
+        Map<String, List<String>> headers = (Map) message.get(Message.PROTOCOL_HEADERS);
+        if (headers != null) {
```
Comment (clockard) [inline platform/security/rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/paos/PaosInInterceptor.java]: not very familiar with this code but If the headers are null should we just create a new map and populate that?

Comment (rzwiefel) [reply platform/security/rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/paos/PaosInInterceptor.java]: yes it absolutely should. i'm going to blame this on vacation brain . Fixed 👍

**Comment by rzwiefel** [comment]
build now

**Comment by rzwiefel** [comment]
some things *changed*..

Previously I was doing this:
screenshot 2018-06-07 18 09 29
screenshot 2018-06-07 18 09 33

But since i am clearing out the headers **anyways**... all i really need is this ( and its safer to not depend on the map coming in anyways):
screenshot 2018-06-07 18 10 34

Thought ya'll would enjoy knowing 😄 Since I committed new code.... Build now

**Comment by clockard** [comment]
Bypassing sonarqube checks since none critical findings were in the change set.

**Pull request merged**